torch/testing/_internal/common_methods_invocations.py

```diff
@@ -220,9 +237,71 @@ def sample_inputs(self, device, dtype, requires_grad=False):
                                         requires_grad=requires_grad)),)
 
 
+# Metadata class for Fast Fourier Transforms in torch.fft.
+class SpectralFuncInfo(OpInfo):
+    """Operator information for torch.fft transforms. """
+
+    def __init__(self,
+                 name,  # the string name of the function
+                 *,
+                 ref=None,  # Reference implementation (probably in np.fft namespace)
+                 dtypes=floating_and_complex_types(),
+                 dtypesIfCPU=None,
+                 dtypesIfCUDA=None,
+                 dtypesIfROCM=None,
+                 ndimensional,
+                 skips=None,
+                 **kwargs):
+        skips = skips if skips is not None else []
+        dtypesIfCPU = dtypesIfCPU if dtypesIfCPU is not None else dtypes
```

Comment:
Are these necessary? Won't the OpInfo class perform this logic?

Reply:
Fixed in #48712. As an aside, does adding small cleanup PRs onto the stack work out better than updating this PR after it's imported?

Reply:
Adding the cleanup to later PRs in the stack (including creating a cleanup PR for later) is easier since editing this requires it be reimported, which will restart the internal test process.